gems/componeer/app/components/componeer/tags/tag_component.rb

```diff
@@ -0,0 +1,24 @@
+module Componeer
+  module Tags
+    class TagComponent < BaseComponent
+      register_as :tag
+
+      attr_reader :text, :type, :size, :color, :shape, :options
+
+      def initialize(text = nil, **options)
+        @text = text
+        @type = options.delete(:type) || :default
```

Comment:
I'd say to we adopt inline method definitions for those options, I'm afraid initializers getting bloated over the time.